drivers/SmartThings/matter-switch/src/eve-energy/init.lua

```diff
@@ -0,0 +1,224 @@
+  -- The powerConsumption report needs to be updated at least every 15 minutes in order to be included in SmartThings Energy
+  -- Eve Energy generally report changes every 10 or 17 minutes
+  local timer = device.thread:call_on_schedule(TIMER_REPEAT, function()
+    requestData(device)
+  end, "polling_schedule_timer")
+
+  device:set_field(RECURRING_POLL_TIMER, timer)
+end
+
+
+-------------------------------------------------------------------------------------
+-- Matter Utilities
+-------------------------------------------------------------------------------------
+
+--- component_to_endpoint helper function to handle situations where
+--- device does not have endpoint ids in sequential order from 1
+--- In this case the function returns the lowest endpoint value that isn't 0
+local function find_default_endpoint(device, component)
+  local res = device.MATTER_DEFAULT_ENDPOINT
+  local eps = device:get_endpoints(nil)
+  table.sort(eps)
+  for _, v in ipairs(eps) do
+    if v ~= 0 then --0 is the matter RootNode endpoint
+      res = v
+      break
+    end
+  end
+  return res
+end
+
+local function component_to_endpoint(device, component_id)
+  -- Assumes matter endpoint layout is sequentional starting at 1.
+  local ep_num = component_id:match("switch(%d)")
+  return ep_num and tonumber(ep_num) or find_default_endpoint(device, component_id)
+end
+
+local function endpoint_to_component(device, ep)
+  local switch_comp = string.format("switch%d", ep)
+  if device.profile.components[switch_comp] ~= nil then
+    return switch_comp
+  else
+    return "main"
+  end
+end
+
+
+-------------------------------------------------------------------------------------
+-- Device Management
+-------------------------------------------------------------------------------------
+
+local function device_init(driver, device)
+  log.info_with({ hub_logs = true }, "device init")
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  device:set_endpoint_to_component_fn(endpoint_to_component)
+  device:subscribe()
+
+  create_poll_schedule(device)
+end
+
+local function device_added(driver, device)
+  -- Reset the values
+  device:emit_event(capabilities.powerMeter.power({ value = 0.0, unit = "W" }))
+  device:emit_event(capabilities.energyMeter.energy({ value = 0.0, unit = "Wh" }))
+end
+
+local function device_removed(driver, device)
+  local poll_timer = device:get_field(RECURRING_POLL_TIMER)
+  if poll_timer ~= nil then
+    device.thread:cancel_timer(poll_timer)
+    device:set_field(RECURRING_POLL_TIMER, nil)
+  end
+end
+
+local function handle_refresh(self, device)
+  requestData(device)
+end
+
+
+-------------------------------------------------------------------------------------
+-- Eve Energy Handler
+-------------------------------------------------------------------------------------
+
+local function matter_handler(driver, device, response_block)
+  log.info(string.format("Fallback handler for %s", response_block))
+end
```

Comment:
this behavior will be inherited from the base driver, so there's no reason to include it here